interfaces/Portal/src/app/services/api.service.ts

```diff
@@ -38,215 +38,128 @@ export class ApiService {
     return headers;
   }
 
+  private async performRequest(
+    method: 'GET' | 'POST' | 'PUT' | 'PATCH' | 'DELETE',
+    url: string,
+    {
+      anonymous = false,
+      body,
+      responseAsBlob = false,
+      isUpload = false,
+      params,
+    }: {
+      anonymous?: boolean;
+      body?: unknown;
+      responseAsBlob?: boolean;
+      isUpload?: boolean;
+      params?: HttpParams;
+    },
+  ) {
+    const security = this.showSecurity(anonymous);
+    console.log(`ApiService ${method}: ${security} ${url}`, body);
+
+    try {
+      const response = await lastValueFrom(
+        this.http
+          .request(method, url, {
+            headers: this.createHeaders(isUpload),
+            responseType: responseAsBlob ? 'blob' : undefined,
+            withCredentials: true,
+            params,
+            body,
+          })
+          .pipe(
+            tap((response) =>
+              console.log(
+                `ApiService ${method}: ${security} ${url}${
+                  params ? `\nParams ${params}` : ''
+                }${body ? `\nBody: ${JSON.stringify(body, null, 2)}` : ''}`,
+                '\nResponse:',
+                response,
+              ),
+            ),
+            catchError((error: HttpErrorResponse): Observable<any> => {
+              return this.handleError(error, anonymous);
+            }),
+          ),
+      );
+      if (response && response.error) {
+        throw response;
+      }
+      return response;
+    } catch (error) {
+      console.error(`Error in ApiService ${method}:`, error);
+      throw error;
+    }
+  }
+
   get(
     endpoint: string,
     path: string,
     anonymous = false,
     responseAsBlob = false,
     params = null,
```

Comment:
To be complete, we could add: `params: HttpParams = null,` (like in the other methods)